lib/cstruct.ml

```diff
@@ -401,6 +402,28 @@ let to_string ?(off=0) ?len:sz t =
      freshly-created value built by [to_bytes t]. *)
   copy t off len
 
+let to_hex_string ?(off=0) ?len:sz t : string =
+  let[@inline] nibble_to_char (i:int) : char =
+    if i < 10 then
+      Char.chr (i + Char.code '0')
+    else
+      Char.chr (i - 10 + Char.code 'a')
+  in
+
+  let len = match sz with None -> length t - off | Some l -> l in
+  if len < 0 || off < 0 || t.len - off < len then
+    err_to_hex_string t off len
+  else (
+    let out = Bytes.create (2 * len) in
+    for i=0 to len-1 do
+      let c = Char.code @@ Bigarray.Array1.get t.buffer (i+off) in
```

Comment:
I think you need to add the offset from the cstruct as well
```suggestion
      let c = Char.code @@ Bigarray.Array1.get t.buffer (t.off+i+off) in
```

Reply:
thanks, good catch. Added a test that fails because of that.